scripts/cmake/CMakeSetup.cmake

```diff
@@ -1,14 +1,22 @@
+# Disallow in-source builds
+if("${PROJECT_SOURCE_DIR}" STREQUAL "${PROJECT_BINARY_DIR}")
```

Comment:
Wouldn't a more general check, e.g. `if ("${PROJECT_BINARY_DIR}/" STRING_STARTSWITH "${PROJECT_SOURCE_DIR}")` be even better. (the search operator is probably not defined.)

Reply:
But this would also exclude `source/build/..` which I find handy. See also comment above.